test/Microsoft.AspNetCore.Mvc.IntegrationTests/ActionParametersIntegrationTest.cs

```diff
@@ -562,5 +596,37 @@ System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator()
                 return GetEnumerator();
             }
         }
+
+        // By default the ComplexTypeModelBinder does not bind to abstract types as we cannot construct them,
+        // but a developer could change this behavior by overridng CreateModel
+        private class CustomComplexTypeModelBinder : ComplexTypeModelBinder
+        {
+            public CustomComplexTypeModelBinder(IDictionary<ModelMetadata, IModelBinder> propertyBinders)
+                : base(propertyBinders)
+            {
+            }
+
+            protected override object CreateModel(ModelBindingContext bindingContext)
+            {
+                if (bindingContext.ModelType == typeof(ClassWithNoDefaultConstructor))
```

Comment:
Instead, `Assert.Equal(typeof(ClassWithNoDefaultConstructor), bindingContext.ModelType)` because binder should never be instantiated for another type.